docs/src/extend/custom-processors.md

```diff
@@ -46,27 +47,27 @@ module.exports = {
 
 **The `preprocess` method** takes the file contents and filename as arguments, and returns an array of code blocks to lint. The code blocks will be linted separately but still be registered to the filename.
 
-A code block has two properties `text` and `filename`; the `text` property is the content of the block and the `filename` property is the name of the block. Name of the block can be anything, but should include the file extension, that would tell the linter how to process the current block. The linter will check [`--ext` CLI option](../use/command-line-interface#--ext) to see if the current block should be linted, and resolve `overrides` configs to check how to process the current block.
+A code block has two properties `text` and `filename`. The `text` property is the content of the block and the `filename` property is the name of the block. The name of the block can be anything, but should include the file extension, which tells the linter how to process the current block. The linter checks the [`--ext` CLI option](../use/command-line-interface#--ext) to see if the current block should be linted and resolves `overrides` configs to check how to process the current block.
 
-It's up to the plugin to decide if it needs to return just one part, or multiple pieces. For example in the case of processing `.html` files, you might want to return just one item in the array by combining all scripts, but for `.md` file where each JavaScript block might be independent, you can return multiple items.
+It's up to the plugin to decide if it needs to return just one part of the non-JavaScript file or multiple pieces. For example in the case of processing `.html` files, you might want to return just one item in the array by combining all scripts. However, for `.md` files, because each JavaScript block might be independent, you can return multiple items.
 
 **The `postprocess` method** takes a two-dimensional array of arrays of lint messages and the filename. Each item in the input array corresponds to the part that was returned from the `preprocess` method. The `postprocess` method must adjust the locations of all errors to correspond to locations in the original, unprocessed code, and aggregate them into a single flat array and return it.
 
 Reported problems have the following location information:
 
 ```typescript
+// TODO: are these the messages?
```

Comment:
Yes, we should probably include the full message interface rather than just the location information here.

Reply:
in latest version i included a typescript definition. you think that's ok?

Reply:
ok, makes sense to me. i think just writing out the TypeScript interface would work. sound good to you?

also, based upon looking at the source code, it seems like there's some properties of this object that we're not documenting here - https://github.com/eslint/eslint/blob/main/lib/shared/types.js#L95 

should those be included in the docs?

Reply:
I'm actually not sure if postprocess receives that full interface or not. @btmills do you know?

I think including the interface as TypeScript is fine.

Reply:
It does! In case it helps, here are a couple examples that I just pulled from linting https://github.com/eslint/eslint-plugin-markdown/tree/ef7dcdccfb94ac7e657a6c66998ce8831f3e58fd/examples/typescript:

```js
[
  {
    ruleId: '@typescript-eslint/ban-types',
    severity: 2,
    message: "Don't use `String` as a type. Use string instead",
    line: 1,
    column: 22,
    nodeType: 'Identifier',
    messageId: 'bannedTypeMessage',
    endLine: 1,
    endColumn: 28,
    fix: { range: [ 21, 27 ], text: 'string' }
  },
  {
    ruleId: '@typescript-eslint/no-explicit-any',
    severity: 1,
    message: 'Unexpected any. Specify a different type.',
    line: 5,
    column: 13,
    nodeType: 'TSAnyKeyword',
    messageId: 'unexpectedAny',
    endLine: 5,
    endColumn: 16,
    suggestions: [
      {
        messageId: 'suggestUnknown',
        fix: { range: [ 82, 85 ], text: 'unknown' },
        desc: 'Use `unknown` instead, this will force you to explicitly, and safely assert the type is correct.'
      },
      {
        messageId: 'suggestNever',
        fix: { range: [ 82, 85 ], text: 'never' },
        desc: "Use `never` instead, this is useful when instantiating generic type parameters that you don't need to know the type of."
      }
    ]
  }
]
```

Reply:
thanks brandon. i will provide a full interface definition here in a markdown list like we do throughout the docs.

however, like we talked about in https://github.com/eslint/eslint/discussions/16557#discussioncomment-4814143, i think well annotated TS definitions and docs generated from them will be great for replacing manually created/maintained reference docs like this.

Reply:
Agreed! It's too easy for the implementation to drift from these docs if they're manual.

Reply:
added the TS type in [239b615](https://github.com/eslint/eslint/pull/16838/commits/239b615eacef0e4b77fb4a6c23d2e1bb1fab5afa). mapped the [JSDoc type](https://github.com/eslint/eslint/blob/main/lib/shared/types.js#L95) to a TS type. 

however this is slightly different from the examples that you included above, @btmills. any idea what might be going on with this?

Reply:
@btmills, following up on the above question

Reply:
We don't (yet) run a type checker on our codebase, so there's currently no automatic enforcement that the implementation does what the type annotations say it will. The JSDoc types are more like documentation at the moment. 

As for what's actually correct, I opened https://github.com/eslint/eslint/issues/16968 to summarize my findings and ask a couple open questions. To enable you to move forward, what you put in 239b615 is good for now. As part of #16968, I can come back and update it to make sure it's fully in sync with whatever changes we make there. But you're right - the real answer is extracting types automatically to prevent drift.

Reply:
thanks for clarifying. will take no further action here.